backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inventory/InventoryRepository.java

```diff
@@ -716,4 +722,55 @@ public Inventory createInventoryLine(@NonNull final InventoryLineCreateRequest r
 
 		return toInventory(inventory);
 	}
+
+	public Set<I_M_InventoryLine> retrieveAllLinesForHU(final @NonNull IContextAware ctxAware, @NonNull final HuId huId)
+	{
+		final List<I_M_HU> includedHUs = huDAO.retrieveIncludedHUs(huDAO.getById(huId));
+
+		final ImmutableSet<HuId> includedHUIds = includedHUs.stream()
+				.map(hu -> HuId.ofRepoId(hu.getM_HU_ID()))
+				.collect(ImmutableSet.toImmutableSet());
+
+		final ImmutableSet<HuId> huIds = ImmutableSet.<HuId>builder()
+				.addAll(includedHUIds)
+				.add(huId)
+				.build();
+
+		final List<I_M_InventoryLine> linesViaHUId = queryBL.createQueryBuilder(I_M_InventoryLine.class)
+				.addInArrayFilter(I_M_InventoryLine.COLUMNNAME_M_HU_ID, huIds)
+				.create()
+				.list();
+
+		final List<I_M_InventoryLine> linesViaHUInventoryLine = queryBL.createQueryBuilder(I_M_InventoryLine_HU.class)
+				.addInArrayFilter(I_M_InventoryLine_HU.COLUMNNAME_M_HU_ID, huIds)
+				.andCollect(I_M_InventoryLine_HU.COLUMNNAME_M_InventoryLine_ID, I_M_InventoryLine.class)
+				.create()
+				.list();
+
+		final ICompositeQueryFilter<I_M_HU_Assignment> filter = queryBL.createCompositeQueryFilter(I_M_HU_Assignment.class)
+				.setJoinOr()
+				.addInArrayFilter(I_M_HU_Assignment.COLUMNNAME_M_HU_ID, huIds)
+				.addInArrayFilter(I_M_HU_Assignment.COLUMNNAME_M_LU_HU_ID, huIds)
+				.addInArrayFilter(I_M_HU_Assignment.COLUMNNAME_M_TU_HU_ID, huIds)
+				.addInArrayFilter(I_M_HU_Assignment.COLUMNNAME_VHU_ID, huIds);
+
+		final List<I_M_InventoryLine> linesViaHUAssignment = queryBL.createQueryBuilder(I_M_HU_Assignment.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_M_HU_Assignment.COLUMNNAME_AD_Table_ID, InterfaceWrapperHelper.getTableId(org.compiere.model.I_M_InventoryLine.class))
+				.filter(filter)
+				.create()
+				.stream()
+				.map(huAssignment -> TableRecordReference.ofReferenced(huAssignment))
+				.sorted(Comparator.comparing(TableRecordReference::getRecord_ID))
+				.distinct()
+				.map(ref -> ref.getModel(ctxAware, I_M_InventoryLine.class))
+				.collect(Collectors.toList());
+
+		return ImmutableSet.<I_M_InventoryLine>builder()
```

Comment:
when adding model classes to sets like this, i'm always a bit uneasy because I have a big amound of FUD towards `hashCode()` & `equals()`: if we load the same `M_InventoryLine` record from DB twice - will the corresponding `I_M_InventoryLine`s be èqual to each other?